tests/test_stubs.py

```diff
@@ -285,6 +285,15 @@ def test_newtype_parameter_annotation(self):
         expected = 'def test(user_id: UserId) -> None: ...'
         assert stub.render() == expected
 
+    def test_union_nonetype_annotation(self):
+        """NoneType should always be rendered as None"""
+        sig = Signature.from_callable(UpdateSignatureHelper.has_annos)
+        sig = update_signature_args(sig, {'a': Dict[str, Dict[str, Union[str, NoneType]]]}, has_self=False,
+                                    ignore_existing_annotations=True)
+        stub = FunctionStub('test', sig, FunctionKind.MODULE)
+        expected = 'def test(a: Dict[str, Dict[str, Union[str, None]]], b) -> int: ...'
```

Comment:
This test is good, though really if the full bug were fixed here, the expected annotation would be `Dict[str, Dict[str, Optional[str]]`

Reply:
Change it when we fix the other bug?  Or just simplify it now?

Reply:
Actually on second thought it would be best if we could adjust this test to use a repro case that doesn't rely on `Union[Something, None]`. Because if we leave it as is, then once #76 is fixed, this test no longer even tests what it's supposed to (avoiding `NoneType`), because there won't be any `None` or `NoneType` here any more, just an `Optional`.

We should be able to repro this with something even simpler like `Dict[str, None]` to keep it totally orthogonal from `Optional` handling.

Reply:
@carljm simplified it.